### ISSUE TITLE
Verify app upgrade data safety

### ISSUE THREAD

**Issue by eb1** [adapt-it/adapt-it-mobile#112]
This is a placeholder issue to go through and verify that AIM can be upgraded without affecting the project and document data.

**Comment by pngbill** [comment]
I just installed the Milestone 0.4.0 candidate and installed it on my 7" Android tablet that I've used for previous AIM version testing. The installation process wiped out my previous project and data, so I would say this is priority item to fix.

**Comment by adaptitbruce** [comment]
On 28/10/2015 7:56 AM, Bill Martin wrote:
>
> I just installed the Milestone 0.4.0 candidate and installed it on my
> 7" Android tablet that I've used for previous AIM version testing. The
> installation process wiped out my previous project and data, so I
> would say this is priority item to fix.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adapt-it/adapt-it-mobile/issues/112#issuecomment-151641045>.
>
Oops, I hadn't read this from Bill when I suggested the announcement
could go out. I agree that this needs to be fixed before anyone trials
it. --Bruce

**Comment by eb1** [comment]
Marking this issue as resolved:

It looks like this issue was due to persisting some of the model items in the browser's local Storage. The fix was to refactor the Book and Project objects so that they used the same sql database we're using for the rest of the objects.

Unfortunately, this isn't a straightforward issue to verify -- it requires two versions starting with the fix from today (d001871):

1. Download and install the latest from the PhoneGap build share site here: https://build.phonegap.com/apps/713362/share. _This will remove your existing data._
2. Wait for the next AIM / PhoneGap build update, then download it and install it over the version you have. I've tried this on my Android tablet, and it retains the data as expected.

What I do _not_ know is if there are different working folders for debug and release versions of AIM. Android seems to have the working directory as hidden unless you have a rooted phone (see https://stackoverflow.com/questions/16791166/location-of-database-in-phone-gap-for-android), so it's unclear whether our tests can safely mix debug and release builds. Fortunately, this is not an end-user concern (they should be working with release versions).

**Comment by pngbill** [comment]
I just downloaded AIM alpha 0.5.0 and attempted to install 0.5.0 while 0.4.5 was installed. My android Samsung 7" tablet (Android 4.4.2) went through the motions of installing, but at the end it said, 
X App not installed
I verified that no newer version was installed by going to the program manager.
I tried this several times but with the same message. So, in the program manager I selected Adapt It Mobile (the 0.4.5 version), and clicked on "Uninstall" (I didn't touch the "Clear" buttons). It uninstalled 0.4.5. Then I went back to my Downloads folder, and double clicked on the 0.5.0 version I had just downloaded. It installed 0.5.0 OK, but running the application, it removed all of my previous data :(. Something is preventing the installation of a newer version of AIM on Android. It seems that the only way to do so is to first uninstall the previous version of AIM which, unfortunately destroys all project data - even though no action was taken to clear any data or cache. I have camera generated screen shots of the whole process available if you want to see them.
